africanus/gridding/wgridder/dask.py

```diff
@@ -0,0 +1,211 @@
+# -*- coding: utf-8 -*-
+
+try:
+    import dask.array as da
+except ImportError as e:
+    dask_import_error = e
+else:
+    dask_import_error = None
+
+import numpy as np
+from africanus.gridding.wgridder.vis2im import DIRTY_DOCS
+from africanus.gridding.wgridder.im2vis import MODEL_DOCS
+from africanus.gridding.wgridder.im2residim import RESIDUAL_DOCS
+from africanus.gridding.wgridder.im2vis import _model_internal as model_np
+from africanus.gridding.wgridder.vis2im import _dirty_internal as dirty_np
+from africanus.gridding.wgridder.im2residim import (_residual_internal
+                                                    as residual_np)
+from africanus.util.requirements import requires_optional
+
+
+def _model_wrapper(uvw, freq, model, freq_bin_idx, freq_bin_counts, cell,
+                   weights, flag, celly, epsilon, nthreads, do_wstacking):
+
+    return model_np(uvw[0], freq, model[0][0], freq_bin_idx, freq_bin_counts,
+                    cell, weights, flag, celly, epsilon, nthreads,
+                    do_wstacking)
+
+
+@requires_optional('dask.array', dask_import_error)
+def model(uvw, freq, image, freq_bin_idx, freq_bin_counts, cell,
+          weights=None, flag=None, celly=None, epsilon=None, nthreads=1,
+          do_wstacking=True):
+    # determine output type
+    complex_type = da.result_type(image, np.complex64)
+
+    # set precision
+    if epsilon is None:
+        if image[0, 0].dtype == np.float64:
```

Comment:
Would it be possible to avoid indexing and use `image.dtype` here ?

Reply:
done